### PR TITLE
Hansp4 branch8

### PR DESCRIPTION
**Pull request by Hajohnson3m**
fix movable block mostly, fixed key spawning location, fixed bug where keys spawn before enemies dies because of spawn clouds, added offset to enemy link collision, made it so collision handler does not check movable block vs immovable block collision

### CODE REVIEW

File: sprint0/sprint0/Collision/LinkBlockCollisionHandler.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ private void HandleMovableBlock(IPlayer link, IBlock block, Direction side)
                     block.Location = new Rectangle(block.Location.X, (int)link.Pos.Y + linkSize, block.Location.Width, block.Location.Height);
                     break;
                 case Direction.e:
-                    block.Location = new Rectangle((int)link.Pos.X + linkSize, block.Location.Y, block.Location.Width, block.Location.Height);
+                    link.Pos += new Vector2(block.Location.Left - (link.Pos.X + linkSize - offset), 0);
```
Comment (Neha968) [inline sprint0/sprint0/Collision/LinkBlockCollisionHandler.cs:75]: Not sure why this change was made

**Review by Neha968 — CHANGES_REQUESTED** [review]

File: sprint0/sprint0/Collision/LinkBlockCollisionHandler.cs (hunk 1 above)
Comment (Hajohnson3m) [reply sprint0/sprint0/Collision/LinkBlockCollisionHandler.cs:75]: because the block should not move to the right

Comment (SnobbyDragon) [reply sprint0/sprint0/Collision/LinkBlockCollisionHandler.cs:75]: ok so do all blocks move up and down? I thought some of them could move side to side. We might need a Property in movable block that says which directions it can move in

File: sprint0/sprint0/RoomLoader/RoomBlocks.cs [hunk 2]
```diff
@@ -57,5 +59,15 @@ public void Draw(SpriteBatch spriteBatch)
             foreach (IBlock block in blocks)
                 block.Draw(spriteBatch);
         }
+
+        public void SwitchToMovableBlock() {
+            IBlock blockToSwitch=null;
+            foreach (IBlock block in blocks) {
+                if (block.GetType().Name=="Block")blockToSwitch=block;
```
Comment (SnobbyDragon) [inline sprint0/sprint0/RoomLoader/RoomBlocks.cs:66]: can use `block is Block`

**Review by SnobbyDragon — CHANGES_REQUESTED** [review]

File: sprint0/sprint0/RoomLoader/RoomBlocks.cs (hunk 2 above)
Comment (Hajohnson3m) [reply sprint0/sprint0/RoomLoader/RoomBlocks.cs:66]: ahah I completele forgot to fix that

**Review by SnobbyDragon — APPROVED** [review]
since we don't care about side to side; approved

**Review by Neha968 — APPROVED** [review]
